### DISCUSSION TITLE
How do you pass an option to increase the request timeout?

### DISCUSSION THREAD

**Discussion by stephenhighemprise** [HoneyryderChuck/httpx#38] [Q&A] [answered]
I am working with a slow service and and am seeing the default timeout stop my request before it's done.  I'd like to increase it to 2 minutes and I've tried a lot of variations of this with and without `timeout` wrapping some of the different timeout option names but I still haven't succeeded in figuring it out from the docs or source code:

`http.post(my_url, json: payload, options: { timeout: { read_timeout: 120 } })`

I just see **NoMethodError** errors such as 

> undefined method `option_options' for an instance of HTTPX::Options

It seems like I might also be able to pass it before that, but I haven't figured it out how to do it here either:

` http = HTTPX.with(headers: header_obj)`

Thanks for your help!

**Comment by HoneyryderChuck** [comment, marked as answer]
As per [the docs](https://gitlab.com/os85/httpx/-/wikis/Timeouts), ":timeout" is top-level, and should not be wrapped under ":options" (in fact, there is no ":options".

```ruby
http.post(my_url, json: payload, timeout: { read_timeout: 120 })

**Reply by stephenhighemprise** [reply]
Thanks!  Somehow I never saw that Wiki webpage and got lost in the Website and rdoc.  :)  Earlier I had been trying to pass `read_timeout: 120` at the top level but forgot to go back and try wrapping that in a `timeout` hash.
